pkg/prometheus/promcfg_test.go

```diff
@@ -11548,6 +11548,176 @@ func TestScrapeClassMetricRelabelings(t *testing.T) {
 	}
 }
 
+func TestScrapeClassAttachAuthorization(t *testing.T) {
```

Comment:
Could we also add tests with a monitor resource defining its own authz and check that it's not overriden?